src/js/utils/language.js

```diff
@@ -61,11 +63,41 @@ export function getLanguage() {
 export const translatedLanguageCodes = ['ar', 'da', 'de', 'es', 'fr', 'it', 'ja', 'nb', 'nl', 'pt', 'ro', 'sv', 'tr', 'zh'];
 
 export function isTranslationAvailable(language) {
-    // 'en' is our default language, therefore it is excluded from the translated language codes.
     return translatedLanguageCodes.indexOf(formatLanguageCode(language)) >= 0;
 }
 
-export function loadJsonTranslation(base, languageCode, oncomplete, onerror) {
-    const url = base + 'translations/' + languageCode + '.json';
-    ajax({ url, oncomplete, onerror, responseType: 'json' });
+export function getCustomLocalization(localization, intl, languageAndCountryCode) {
+    localization = localization || {};
+    intl = intl || {};
+    const languageCode = formatLanguageCode(languageAndCountryCode);
+    const languageAndCountryCustomization = languageCode === languageAndCountryCode ? {} :
+        intl[languageAndCountryCode] || intl[languageAndCountryCode.toLowerCase()] ||
+        intl[languageAndCountryCode.replace('-', '_')] || intl[languageAndCountryCode.toLowerCase().replace('-', '_')];
```

Comment:
This should be simplified. Only `toLowerCase()` once, and only  `replace('-', '_')` once.

Another way to do it that may be safer is to normalize `languageAndCountryCode`AND the properties in `intl` so they both are lower case and only use underscore. Do there here to find a match regardless of case or separator:

```js
const intlNormalized = Object.keys(intl).reduce((obj, key) => {
  obj[key.toLowerCase().replace('-', '_')] = intl[key];
  return obj;
},  {});
const langAndCCNormalized = langAndCCNormalized.toLowerCase().replace('-', '_');
const languageAndCountryCustomization = intlNormalized[langAndCCNormalized];
```